docs/adrs/024-remote-module-invocation.md

```diff
@@ -0,0 +1,280 @@
+module. One issue possible issue for the future is that the RemoteModelInstance relies on 
+dataclasses created during service generation.
+
+
+
+### RemoteModelFinder
+
+
+```yaml
+model_management:
+    finders:
+        <finder name>:
+            type: REMOTE
+            config:
+                connection:
+                    host: str <remote host>
+                    port: int <remote port>
+                    protocol: Optional[str]="grpc" <protocol the remote server is using (grpc or http)>
+                    tls:
+                        enabled: Optional[bool]=False <if ssl is enabled on the remote server>
+                        ca: Optional[str]=None <path to remote ca file>
+                        cert: Optional[str]=None <path to MTLS cert>
+                    options:  Optional[Dict[str,str]]={} <optional dict of grpc or http configuration options>
+                discover_models: Optional[bool]=True <bool to automatically discover remote models via the /info/models endpoint> 
+                supported_models: Optional[Dict[str, str]]={} <mapping of model names to module_ids that this remote supports>
+                    <model_path>: <module_id>
+```
+
+The proposed configuration for the RemoteModelFinder is above. The only required field is the 
+generic `connection` dictionary that supports a secure channel, mutual TLS, and custom GRPC/HTTP 
+options. The `connection.host` setting contains the remote's hostname, while `connection.port` determines the 
+runtime port. The optional `connection.protocol` config is used to select which protocol to send
+requests over, with the default being `grpc`. The `connection.tls` dictionary contains all information 
+related to TLS with `tls.enabled` controlling if the server is running SSL, `tls.ca` is the path to
+the CA file that the remote's certificate is signed by, and finally, `tls.cert` is the path to the
+MTLS client certificate, which will be sent with the request. The final connection config is `connection.options` 
+which defines a list of options to pass to either the HTTP or GRPC request; for an example of options,
+take a look at the [GRPC Channel options](https://grpc.github.io/grpc/core/group__grpc__arg__keys.html#details)
+
+
+Two additional optional fields help control what models this remote supports. The 
+`discover_models` setting is a boolean that controls if the finder should query the remote runtime
+to dynamically discover what models are loaded and their corresponding `module_id`'s. The
+`supported_models` config is a dictionary that contains a static mapping of model_paths to module_ids 
+that the remote supports. The `supported_models`  setting is required to add support for remotes that
+don't have a reflection api or ones that lazily load their models (like ModelMesh). 
+
+
+To help illustrate the above config, we included some pseudo python code to illustrate what happens 
```

Comment:
>  I think it would be possible to include these configurations with a TGISAutoFinder
What do you mean by this? Like move the TGIS's connection info from the Finder to the backend?

> the RemoteModelFinder only returned non-None for models that are actually supported in the remote 
Yep that was the idea! The `RemoteModelFinder` will only return a `RemoteModuleConfig` if it's l>ocated both the remote model and local ModuleBase.

Reply:
I was just trying to figure out in my head if `TGISAutoFinder` and `RemoteModelFinder` could co-exist. I think they can!